src/Spatial/Euclidean/ILine2D.cs

```diff
@@ -44,6 +44,6 @@ public interface ILine : IEquatable<ILine>
         /// </summary>
         /// <param name="line">the line to be compared against</param>
         /// <returns>True if the lines are parallel</returns>
-        bool IsParallel(ILine line);
+        bool IsParallel<T>(T line) where T : ILine2D;
```

Comment:
I still think `bool IsParallel<T>(T line, Angle tolerance)` make sense. But we can wait and see what other people have to say.